### DISCUSSION TITLE
Possible to obtain edges chosen by graclus?

### DISCUSSION THREAD

**Discussion by cameronosmith** [pyg-team/pytorch_geometric#2430] [Q&A] [answered]
Hi, is there a way I can obtain the indices of the edges that graclus chooses?

Also, is EdgePool basically graclus clustering but with predicting the edge weights first?

**Comment by rusty1s** [comment, marked as answer]
`graclus` does not choose edges, it will cluster nodes. The resulting adjacency matrix is then given by the union of edges of clustered nodes.

`EdgePooling` is different in that it will compute a score for each edge, and will contract edges according to that score.

**Reply by cameronosmith** [reply]
But doesn't graclus pick which nodes to cluster based on the largest edge weight possible in the node's available neighborhood? In that sense isn't it like EdgePool?

**Reply by rusty1s** [reply]
Not entirely. `graclus` still iterates over nodes randomly, and picks the neighbor with highest edge weight to form a cluster for each random node, while `EdgePool` will globally sort edge scores and visit them in order.

**Reply by cameronosmith** [reply]
Right that's what I figured. But like EdgePool does in multiplying the contracted nodes by their contracted edge's score, is there a way to collect the indices of the edges chosen by graclus? Or should I just make a symmetric edge score and collect nodes in the same cluster to recompute the edge score to enable the edge score gradient to flow through them?

**Reply by rusty1s** [reply]
I see. If you are interested in the "edges" chosen by `graclus`, you can directly look for this in the `cluster` vector, i.e., two nodes will have the same index if they are clustered together. These pairs of nodes will denote the edges chosen by `graclus`.
